recipes/go/build.sh

```diff
@@ -0,0 +1,24 @@
+# First, build go1.4 using gcc, then use that go to build go>1.4
+# Should I make this a separate package called go-bootstrap?
+# Should I check the checksum of this tarball? What about windows?
+mkdir go-bootstrap && pushd $_
```

Comment:
Had asked this same question before. Really we need `conda-build` to install executable only dependencies. That way we can build `go` 1.4 first and then make it an executable only requirement for later versions of `go`. Some people accomplish this by using `conda` to create another environment for use in the build, but it is really an ugly hack.

Had raised issue ( https://github.com/conda/conda-build/issues/1467 ) for bootstrapping compilers (with `go` in mind in particular). Though am unclear how to tackle this with the proposed fix, which is likely due to my own lack of familiarity with the solution. Maybe @msarahan can advise?

Reply:
There is one instance of using cmake-binary as build dependency, but in that scenario, a separate conda package was created, and since it's name was different, it could be added as a build dependency. That's why I had asked whether we should be using go-bootstrap as a separate package where go1.4 would be built using our toolchain.

Reply:
The host/build section hack which Michael mentioned can only be used with CB 3. And CF is not there yet. :thinking: